### PR TITLE
6632 - Datepicker Increment/Decrement Settings

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR will add a setting for to allow keyboard hotkeys for increment and decrement of days in datepicker.

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise/issues/6632
Closes https://github.com/infor-design/enterprise/issues/6653

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Go to http://localhost:4000/components/datepicker/test-datepicker-today-increment-key.html
- In the first date field, press `t` to enter todays date
- While in the date field, press the `+` key
- The date should be incremented
- While in the date field, press the `-` key
- The date should be decremented

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by yohannahbautista — APPROVED** [review]

**Review by ericangeles — DISMISSED** [review]

File: src/components/datepicker/datepicker.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ const COMPONENT_NAME = 'datepicker';
  * @param {string} [settings.attributes] Add extra attributes like id's to the element. For example `attributes: { name: 'id', value: 'my-unique-id' }`
  * @param {string} [settings.autocompleteAttribute="off"] Allows prevention of built-in browser typeahead by changing/removing an `autocomplete` attribute to the field.
  * @param {boolean} [settings.tabbable=true] If true, causes the Datepicker's trigger icon to be focusable with the keyboard.
+ * @param {boolean} [settings.updateWithKeyboard=true] If true, enables increment/decrement of day by using +/- in the keyboard.
```
Comment (tmcconechy) [inline src/components/datepicker/datepicker.js:93]:
```suggestion
 * @param {boolean} [settings.updateWithKeyboard=false] If true, enables increment/decrement of day by using +/- in the keyboard.
```

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Something a little off here

a) what should the default be? (im fine with it off but should go in "important changes"
b) http://localhost:4000/components/datepicker/example-index.html you can still press "t" but not +/-
c) but http://localhost:4000/components/datepicker/test-increment-decrement-day.html does work as advertised

**Review by tmcconechy — COMMENTED** [review]
Maybe we could make two settings: `incrementWithKeyboard` `todayWithKeyboard` noting that https://github.com/infor-design/enterprise/issues/6653 is very similar (having a hard idea with the name so open to suggestions).

cc @EdwardCoyle

**Comment by tmcconechy** [comment]
@tjamesallen15 Can we go ahead and fix https://github.com/infor-design/enterprise/issues/6653 as well with the additional today setting for "t"? Seems closely related

**Comment by tjamesallen15** [comment]
> @tjamesallen15 Can we go ahead and fix #6653 as well with the additional today setting for "t"? Seems closely related

Sure. Will do!

**Review by tmcconechy — APPROVED** [review]
Nice 👍🏻

**Review by yohannahbautista — APPROVED** [review]

**Pull request merged**